driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedRetryableWritesTest.java

```diff
@@ -24,12 +24,29 @@
 import java.net.URISyntaxException;
 import java.util.Collection;
 
+import static com.mongodb.ClusterFixture.isDiscoverableReplicaSet;
+import static com.mongodb.ClusterFixture.isSharded;
+import static com.mongodb.ClusterFixture.serverVersionLessThan;
+import static org.junit.Assume.assumeFalse;
+
 public class UnifiedRetryableWritesTest extends UnifiedSyncTest {
     public UnifiedRetryableWritesTest(@SuppressWarnings("unused") final String fileDescription,
                                       @SuppressWarnings("unused") final String testDescription,
                                       final String schemaVersion, final BsonArray runOnRequirements, final BsonArray entitiesArray,
                                       final BsonArray initialData, final BsonDocument definition) {
         super(schemaVersion, runOnRequirements, entitiesArray, initialData, definition);
+        customSkips(testDescription);
+    }
+
+    public static void customSkips(final String description) {
+        // Remove all this as part of JAVA-5125
+        if (isSharded() && serverVersionLessThan(5, 0)) {
+            assumeFalse(description.contains("succeeds after WriteConcernError"));
+            assumeFalse(description.contains("succeeds after retryable writeConcernError"));
+        }
+        if (isDiscoverableReplicaSet() && serverVersionLessThan(4, 4)) {
+            assumeFalse(description.equals("RetryableWriteError label is added based on writeConcernError in pre-4.4 mongod response"));
```

Comment:
This is actually not one of the converted tests.  Rather, it's a new unified test that was added independently and is in our backlog to add.  I'll update the description of that ticket to indicate the test has been merged but is being skipped.

Given that this is 4.2, it's unlikely we'll ever invest in fixing this.